src/AppInstallerCLICore/Commands/SettingsCommand.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "SettingsCommand.h"
+#include "Workflows/WorkflowBase.h"
+#include "Resources.h"
+#include <winget/UserSettings.h>
+
+namespace AppInstaller::CLI
+{
+    using namespace Utility::literals;
+    using namespace AppInstaller::Settings;
+
+    using namespace std::string_view_literals;
+
+    std::vector<Argument> SettingsCommand::GetArguments() const
+    {
+        return {};
+    }
+
+    Resource::LocString SettingsCommand::ShortDescription() const
+    {
+        return { Resource::String::SettingsCommandShortDescription };
+    }
+
+    Resource::LocString SettingsCommand::LongDescription() const
+    {
+        return { Resource::String::SettingsCommandLongDescription };
+    }
+
+    std::string SettingsCommand::HelpLink() const
+    {
+        return "https://aka.ms/winget-settings";
+    }
+
+    void SettingsCommand::ExecuteInternal(Execution::Context& context) const
+    {
+        // Show warnings only when the setting command is executed.
+        for (const auto& warning : Settings::User().GetWarnings())
+        {
+            context.Reporter.Warn() << warning << std::endl;
+        }
+
+        User().PrepareToShellExecuteFile();
+
+        std::string filePathUTF8Str = UserSettings::SettingsFilePath().u8string();
+
+        // Some versions of windows will fail if no file extension association exists, other will pop up the dialog
+        // to make the user pick their default.
+        // Kudos to the terminal team for this work around.
+        HINSTANCE res = ShellExecuteA(nullptr, nullptr, filePathUTF8Str.c_str(), nullptr, nullptr, SW_SHOW);
```

Comment:
>ShellExecuteA [](start = 24, length = 13)

I don't think we have done anything to set our process to UTF-8 defaults, enabling UTF-8 *A apis.  So this should probably be use the W function.  Or potentially figure out how we do set ourself as UTF-8 aware (although I suspect that older OS versions would still need to use W).

Reply:
Changed. We are using ShellExecuteExA in ShellExecuteInstallerHandler.cpp, is that expected?

---
In reply to: [443074390](https://github.com/microsoft/winget-cli/pull/436#discussion_r443074390) [](ancestors = 443074390)

Reply:
That has its own issue, and yes, it should be changed (but not in the PR).

---
In reply to: [443093477](https://github.com/Microsoft/pkg/pull/436#discussion_r443093477) [](ancestors = 443093477,443074390)